pkg/registry/minion/rest.go

```diff
@@ -143,15 +144,17 @@ func ResourceLocation(getter ResourceGetter, connection client.ConnectionInfoGet
 	node := nodeObj.(*api.Node)
 	host := node.Name // TODO: use node's IP, don't expect the name to resolve.
 
-	if portReq != "" {
-		return &url.URL{Host: net.JoinHostPort(host, portReq)}, nil, nil
-	}
-
 	scheme, port, transport, err := connection.GetConnectionInfo(host)
 	if err != nil {
 		return nil, nil, err
 	}
 
+	if strconv.Itoa(ports.KubeletPort) == portReq {
+		return &url.URL{Scheme: scheme, Host: net.JoinHostPort(host, portReq)}, transport, nil
+	} else if portReq != "" {
+		return &url.URL{Host: net.JoinHostPort(host, portReq)}, nil, nil
+	}
+
```

Comment:
can you replace the end of this function (from line 147 on) with this or something like it? I think it's a lot clearer.

```
if portReq == "" || strconv.Itoa(ports.KubeletPort) == portReq {
	scheme, port, transport, err := connection.GetConnectionInfo(host)
	if err != nil {
		return nil, nil, err
	}
	return &url.URL{
			Scheme: scheme,
			Host: net.JoinHostPort(
				host,
				strconv.FormatUint(uint64(port), 10),
			),
		},
		transport,
		nil
}
return &url.URL{Host: net.JoinHostPort(host, portReq)}, nil, nil
```